content/en/blog/_posts/2024-07-18-kubernetes-1.31-deprecations-and-removals.md

```diff
@@ -0,0 +1,116 @@
+
+Given Go's [compatibility policy for GODEBUGs](https://go.dev/blog/compat), the `x509sha1` GODEBUG and consequently the support for SHA-1 certificates will [fully go away in go1.24](https://tip.golang.org/doc/go1.23) which would release in the first half of 2025. If you are reliant on SHA-1 certificates, please start moving off of them.
+
+Please see [Kubernetes issue #125689](https://github.com/kubernetes/kubernetes/issues/125689) to get a better idea of timelines around the support for SHA-1 going away, when Kubernetes releases plans to adopt go1.24, and for more details on how to detect usage of SHA-1 certificates via metrics and audit logging. 
+
+## Deprecations and Removals in Kubernetes 1.31
+
+### Deprecate kustomize from kubectl [KEP 4706](https://github.com/kubernetes/enhancements/issues/4706)
+In the early days of Kubernetes, when not that many tools in the field of declarative configurations existed, [kustomize](https://github.com/kubernetes-sigs/kustomize) was introduced and to make life easier for kubectl users it was included in kubectl, by default.
+
+Over the past several years the ecosystem around Kubernetes grew significantly, resulting in wider spread of tooling, including declarative configuration. With these new options, it's best left to the user to match their use cases with the best tool capable of resolving the problem at hand. Using [kubectl plugins](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/), users can pick and choose the best tool to match their needs.
+
+With Kubernetes v1.31, kustomize will be deprecated from kubectl. It will be removed in a future release. This will allow both tools to be developed and maintained separately.
+
+### Deprecate Kubelet RunOnce mode [KEP 4580](https://github.com/kubernetes/enhancements/issues/4580)
+The kubelet support for RunOnce mode is being deprecated and will be removed in a future release. Users are advised to deprecate RunOnce mode by disabling the `LegacyNodeRunOnceMode` feature gate. Starting in Kubernetes 1.31, kubelet will refuse to start if this feature gate is disabled. 
+
+If you still rely on this feature, migrate to using the `podman kube` subcommand. Refer to the [podman documentation](https://docs.podman.io/en/latest/markdown/podman-kube.1.html.) for more information.
+
+### Deprecate status.nodeInfo.kubeProxyVersion field [KEP 4004](https://github.com/kubernetes/enhancements/issues/4004)
+The `status.nodeInfo.kubeProxyVersionv1.Node` field is being deprecated due to inaccuracies and will be removed in a future release. This field is set by the kubelet, which does not have reliable information about the kube-proxy version or whether kube-proxy is running. 
+
+After deprecation, users will no longer be able to retrieve the kube-proxy version from the Node object.
+
+### Removal of in-tree cloud providers - sig-cloudprovider
```

Comment:
@elmiko Not sure if you are the right person but can you get someone in the sig-cloudprovider to review? Thanks a lot!

Reply:
thanks for the ping @mbianchidev , this looks accurate to me.

cc @bridgetkromhout ptal also